### PR TITLE
Run hedera-cryptography-altbn128 tests on multiple operating systems

### PR DESCRIPTION
**Pull request by jjohannes**
**Description**:

- Follow up to #70 
- Affected by #81

### CODE REVIEW

**Comment by jjohannes** [comment]
The CI build sometimes runs into this problem with `xwin`: https://github.com/Jake-Shadle/xwin/issues/141

If it fails, trigger a re-run. I hope there will be a fix or stable workaround soon.

File: gradle/plugins/src/main/kotlin/com/hedera/gradle/tasks/CargoBuildTask.kt [hunk 1]
```diff
@@ -85,11 +85,13 @@ abstract class CargoBuildTask : DefaultTask() {
 
     private fun installCargoCrossCompiler(buildsForWindows: Boolean) {
         exec.exec {
-            val crossCompiler = if (buildsForWindows) "xwin" else "cargo-zigbuild"
+            val crossCompiler = if (buildsForWindows) "xwin@0.6.6-rc.1" else "cargo-zigbuild"
```
Comment (rbarker-dev) [inline gradle/plugins/src/main/kotlin/com/hedera/gradle/tasks/CargoBuildTask.kt]: Can you add a comment linking to the thread where it was determined we needed to use 0.6.6-rc.1 here in the code? That way if a more stable version is released it should be a little easier to justify switching to it.

**Review by rbarker-dev — CHANGES_REQUESTED** [review]

File: gradle/plugins/src/main/kotlin/com/hedera/gradle/tasks/CargoBuildTask.kt (hunk 1 above)
Comment (jjohannes) [reply gradle/plugins/src/main/kotlin/com/hedera/gradle/tasks/CargoBuildTask.kt]: :+1: Comment added. I am following the _xwin_ issue and will keep an eye on this to remove it again once it is no longer needed.

File: common/hedera-common-nativesupport/src/main/java/com/hedera/common/nativesupport/NativeLibrary.java [hunk 2]
```diff
@@ -252,7 +252,10 @@ private void installUnchecked(@NonNull final Class<?> c) {
     @SuppressWarnings("ResultOfMethodCallIgnored")
     private void install(@NonNull final InputStream resourceStream) throws IOException {
         Objects.requireNonNull(resourceStream, "resourceStream must not be null");
-        final String fileName = Path.of(name).getFileName().toString();
+        final OperatingSystem os = OperatingSystem.current();
+        final String libName =
+                os == OperatingSystem.WINDOWS ? name + "." + libExtensions.get(OperatingSystem.WINDOWS) : name;
+        final String fileName = Path.of(libName).getFileName().toString();
```
Comment (jjohannes) [inline common/hedera-common-nativesupport/src/main/java/com/hedera/common/nativesupport/NativeLibrary.java:258]: @timo0 now that the tests are running on Windows, I found this other special case: On Windows, the DLL extension is part of the library name to load.

On the upside, with this change the tests pass on Windows. Which is a good sign that the cross-compilation is successful. 🎉

Comment (mxtartaglia-sl) [reply common/hedera-common-nativesupport/src/main/java/com/hedera/common/nativesupport/NativeLibrary.java:258]: thanks @jjohannes

**Review by mxtartaglia-sl — DISMISSED** [review]

File: .github/workflows/build.yml [hunk 3]
```diff
@@ -82,5 +82,41 @@ jobs:
       - name: Publish Unit Test Results
         uses: step-security/publish-unit-test-result-action@4519d7c9f71dd765f8bbb98626268780f23bab28 # v2.17.0
         with:
-          files: |
-            test-results/**/*.xml:
+          files: "**/build/test-results/test/TEST-*.xml"
```
Comment (jjohannes) [inline .github/workflows/build.yml:85]: Not related directly, but the path here was wrong and thus test results were not published to PRs.

**Review by mxtartaglia-sl — APPROVED** [review]

File: .github/workflows/build.yml (hunk 3 above)
Comment (mxtartaglia-sl) [reply .github/workflows/build.yml:85]: thanks!

**Review by rbarker-dev — APPROVED** [review]
LGTM

**Pull request merged**